snapcraft/cli/lifecycle.py

```diff
@@ -74,6 +75,11 @@ def _execute(  # noqa: C901
     if build_provider in ["host", "managed-host"]:
         apply_host_provider_flags(build_provider_flags)
         project_config = project_loader.load_config(project)
+        if is_managed_host:
+            # Managed hosts repositories are configured with defaults
+            # set by snapcraft.
+            repo.Repo.initialize_snapcraft_defaults()
```

Comment:
Will this mean that if I manually do things, the next time I run, those settings will be overwritten? Maybe stamp to not constantly reinitialize?